tests/unit/s2n_tls_prf_test.c

```diff
@@ -105,6 +105,41 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(conn));
     }
 
+    /* s2n_prf_calculate_master_secret */
+    {
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+
+        conn->secure.cipher_suite = &s2n_ecdhe_ecdsa_with_aes_256_gcm_sha384;
+
+        EXPECT_MEMCPY_SUCCESS(conn->secrets.rsa_premaster_secret, premaster_secret_in.data, premaster_secret_in.size);
+        EXPECT_MEMCPY_SUCCESS(conn->secrets.client_random, client_random_in.data, client_random_in.size);
+        EXPECT_MEMCPY_SUCCESS(conn->secrets.server_random, server_random_in.data, server_random_in.size);
+
+        struct s2n_blob pms = {0};
+        EXPECT_SUCCESS(s2n_blob_init(&pms, conn->secrets.rsa_premaster_secret, sizeof(conn->secrets.rsa_premaster_secret)));
+
+        /* Errors when handshake is not at the Client Key Exchange message */
+        EXPECT_FAILURE_WITH_ERRNO(s2n_prf_calculate_master_secret(conn, &pms), S2N_ERR_SAFETY);
+
+        /* Advance handshake to Client Key Exchange message */
+        conn->handshake.handshake_type = NEGOTIATED | FULL_HANDSHAKE;
+        while(ACTIVE_MESSAGE(conn) != CLIENT_KEY) {
+            conn->handshake.message_number++;
+        }
+
+        /* Master secret is calculated when handshake is at Client Key Exchange message*/
+        EXPECT_SUCCESS(s2n_prf_calculate_master_secret(conn, &pms));
+        EXPECT_EQUAL(memcmp(conn->secrets.master_secret, master_secret_in.data, master_secret_in.size), 0);
+
+        conn->ems_negotiated = true;
+        EXPECT_SUCCESS(s2n_prf_calculate_master_secret(conn, &pms));
+
+        /* Prove that the extended master secret calculated is different than the master secret calculated */
+        EXPECT_NOT_EQUAL(memcmp(conn->secrets.master_secret, master_secret_in.data, master_secret_in.size), 0);
+
```

Comment:
If s2n_prf_calculate_master_secret is called twice without ems_negotiated, do the secrets produced match?

Reply:
The unit test above proves that s2n_prf_calculate_master_secret will always calculate the same secret if given the same inputs.